kernel/src/lib.rs

```diff
@@ -340,6 +344,25 @@ pub trait ExpressionHandler: AsAny {
         expression: Expression,
         output_type: DataType,
     ) -> Arc<dyn ExpressionEvaluator>;
+
+    /// Create a single-row null-value [`EngineData`] of a single column of type `output_type`.
+    fn null_row(&self) -> DeltaResult<(Box<dyn EngineData>, SchemaRef)>;
+
+    /// Create a single-row [`EngineData`] by applying the given schema to the leaf-values given in
+    /// `values`.
+    // Note: we will stick with a Schema instead of DataType (more constrained can expand in
+    // future)
+    fn create_one(&self, schema: SchemaRef, values: &[Scalar]) -> DeltaResult<Box<dyn EngineData>> {
```

Comment:
Do we ever expect engines to override this with their own implementation?

If not, we should remove it from the trait and just make it a `pub(crate)` function somewhere.

Reply:
hm that's a good point - I could conceive of an engine that would prefer to 'build' their own arrays like in my old implementation instead of doing the new_null and expression evaluation, but perhaps we should just make this pub(crate) for now and let that optimization occur if engines ask for it? I guess the downside of the current approach is just 'leaking' extra code in pub interface?

Reply:
Yeah, I'd keep this `pub(crate)` for now if we don't know we need it. We can always move it to the trait if someone asks but it's much harder to the the other way.

Reply:
sure sounds good. i moved to a separate 'extension' trait - lmk if you think of a better way to implement it. (and unfortunately rustc keeps warning that it isn't used even though it is - if i remove it doesn't compile lol - so i added `allow(dead_code)` to get it to shut up and can make an issue to try to sort that out later?)

Reply:
Hrmm, odd. Yeah let's make an issue just so we don't completely forget it

Reply:
yea will do - was getting annoyed with it so decided to not go further down that rabbit hole hahaha

Reply:
https://github.com/delta-io/delta-kernel-rs/issues/746